### ISSUE TITLE
LoRa realisation

### ISSUE THREAD

**Issue by v1docq** [v1docq/FedCore#3]
**Ссылки на постановку задачи и существующие реализации алгоритмов**

1.https://github.com/fxmeng/peft/blob/main/src/peft/tuners/lora/layer.py#L168
2⚡️ PiSSA: Principal Singular Values and Singular Vectors Adaptation of Large Language Models

Significantly improved finetuned perf by simply changing the initialization of LoRA's AB matrix from Gaussian/zero to principal components. (http://t.me/data_analysis_ml)

On GSM8K, Mistral-7B fine-tuned with PiSSA achieves an accuracy of 72.86%, outperforming LoRA’s 67.7% by 5.16%.

▪️Github: https://github.com/GraphPKU/PiSSA
▪️Paper: https://arxiv.org/abs/2404.02948
3. Что делать? Реализовать эти методы в рамках нашего фреймворка

**Ссылки на теорию и что надо будет реализовать**
1.Исходник учебника на английском - [RMT4ML.pdf](https://github.com/v1docq/FedCore/files/15108435/RMT4ML.pdf)

2. Краткий конспект по главам - [Теория случайных матриц для машинного обучения.pdf](https://github.com/v1docq/FedCore/files/15108433/default.pdf). Надо расширить и дополнить формулами.
3. Какие главы читать? Introduction,  Random Matrix Theory, Kernel Methods, Large Networks. Каждая глава - конспект с формулами. На 1 главу - 2 недели (дедлайн). НЕ НАДО ПЫТАТЬСЯ ПОНЯТЬ КАЖДУЮ ФОРМУЛУ, НЕ НАДО ПЕРЕСКАКИВАТЬ ГЛАВЫ.

**Что делать?**

2.Random Matrix Theory - что такое резольвента, закон Марченко-Пастура
3.Из главы Kernel Methods - разобраться что такое функция ядра, в чем идея и польза. Что такое гауссово ядро, как выполнять нормирование данных, что такое случайное ядро. 
4. Из главы Large Networks - реализовать архитектуры "случайных" нейронок для задач регрессии и классификации (ссылки на код там есть) посмотреть связь между ядром апроксимации и функциями активации
